src/main/java/bio/terra/app/logging/PerformanceLoggerON.java

```diff
@@ -0,0 +1,128 @@
+ *     a value for this new field.
+ *   - Specify the new information as a serializable object (e.g. String or property of a POJO) and pass it to the
+ *     existing log methods with the additionalInfo parameter.
+ *  New types of information that seem broadly useful should use the first option and those that will probably only
+ *  be used in a small number of places should use the second option.
+ */
+public class PerformanceLoggerON implements PerformanceLogger {
+
+    public static final boolean isPerformanceLoggingEnabled = true;
+
+    private ObjectMapper objectMapper;
+
+    private static Logger logger = LoggerFactory.getLogger(PerformanceLoggerON.class);
+    private static final String PerformanceLogFormat =
+        "TimestampUTC: {}, JobId: {}, Class: {}, Operation: {}, " +
+        "ElapsedTime: {}, IntegerCount: {}, AdditionalInfo: {}";
+
+    // Implementation choice note:
+    // Using a thread local map for timers instead of a static concurrent hash map avoids locks when updating the map.
+    // Thread local means we can't track timers across different threads, while a concurrent hash map would allow this.
+    // Neither option allows tracking timers across different DR Managers. We could use SQL to calculate time durations
+    // between start/end timestamps to address that use case.
+    private static final ThreadLocal<Map<String, Instant>> startTimeMap =
+        ThreadLocal.withInitial(() -> new HashMap<>());
+
+    @Autowired
+    public PerformanceLoggerON(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+        logger.info("Performance logging ON");
+    }
+
+    public void log(String jobId, String className, String operationName,
+                    long elapsedTime, long integerCount, Serializable additionalInfo) {
+        // serialize the additional info object to JSON
+        String additionalInfoStr = "";
+        if (additionalInfo != null) {
+            try {
+                additionalInfoStr = objectMapper.writeValueAsString(additionalInfo);
+            } catch (JsonProcessingException jsonEx) {
+                // if we hit a serialization error with the additional information object, use an empty string
+                // and log the error separately for debugging
+                logger.info("Failure serializing additionalInfo to JSON. " + additionalInfo.toString());
+                additionalInfoStr = "";
+            }
+        }
+
+        logger.info(PerformanceLogFormat, currentTimestamp(), jobId, className, operationName,
+            elapsedTime, integerCount, additionalInfoStr);
+    }
+
+    /**
+     * Generate a UTC timestamp for the current instant.
+     */
+    private String currentTimestamp() {
+        return Instant.now().atZone(ZoneOffset.UTC).format(DateTimeFormatter.ISO_INSTANT);
+    }
+
+    public String timerStart() {
+        // generate a timer reference id to use for tracking
+        String timerId = UUID.randomUUID().toString();
```

Comment:
That is a reasonable expensive operation. Could the id be a composite of the thread id and a counter rather than a UUID?

Reply:
Yes, done.